scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1516,6 +1536,152 @@
+          ; so that we can treat them differently throughout the scheduler.
+          :pod-labels {"twosigma.com/scheduler" "kubernetes"}
+
+          ; Cook has a "novel host constraint", which disallows a job from
+          ; running on the same host twice. So, we need to avoid running a
+          ; real job pod on any of the hosts that it won't be able to run on.
+          :pod-hostnames-to-avoid (constraints/job->previous-hosts-to-avoid job)
+          :task-request {:scalar-requests (walk/stringify-keys pool-specific-resources)
+                         :job {:job/pool {:pool/name pool-name}
+                               :job/environment environment
+                               :job/name name
+                               :job/user user
+                               :job/uuid uuid}
+                         ; Need to pass in resources to task-metadata->pod for gpu count
+                         :resources pool-specific-resources}})))))
+
+(defn handle-kubernetes-scheduler-pool
+  "Handle scheduling pending jobs using the Kubernetes Scheduler."
+  [conn pool-name->pending-jobs-atom pool-name scheduler-config compute-clusters
+   job->acceptable-compute-clusters-fn user->quota user->usage-future mesos-run-as-user]
+  (log-structured/info "Running handler for Kubernetes Scheduler pool" {:pool pool-name})
+  (try
+    (let [user->usage (tracing/with-span [s {:name "scheduler.kubernetes-handler.resolve-user-to-usage-future"
+                                             :tags {:pool pool-name :component tracing-component-tag}}]
+                        @user->usage-future)
+          {max-considerable :max-jobs-considered} scheduler-config
+          db (db conn)
+          pending-jobs (get @pool-name->pending-jobs-atom pool-name)
+          ;; We need to filter pending jobs based on quota so that we don't
+          ;; submit beyond what users have quota to actually run.
+          jobs (prom/with-duration
+                 prom/scheduler-handle-resource-offers-pending-to-considerable-duration {:pool pool-name}
+                 (timers/time!
+                  (timers/timer (metric-title "kubernetes-handler-considerable-jobs-duration" pool-name))
+                  (pending-jobs->considerable-jobs
+                   db pending-jobs user->quota user->usage max-considerable pool-name)))
+          job-uuids (set (map :job/uuid jobs))]
+      (log-structured/info "Considering jobs" {:pool pool-name :number-considered-jobs (count job-uuids)})
+      (if (seq jobs)
+        (do
+          (swap! pool-name->pending-jobs-atom
+                 remove-matched-jobs-from-pending-jobs
+                 job-uuids pool-name)
+          (log-structured/debug (print-str "Updated pool-name->pending-jobs-atom:" (get @pool-name->pending-jobs-atom pool-name))
+                                {:pool pool-name})
+          (let [autoscaling-compute-clusters (filter #(cc/autoscaling? % pool-name) compute-clusters)
```

Comment:
This block of code has a high overlap with a block inside of `trigger-autoscaling!` can you refactor out the common components?

Reply:
Done.